etc/scripts/freeze_and_update_reqs.py

```diff
@@ -0,0 +1,135 @@
+
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+
+import fnmatch
+import hashlib
+import io
+import os
+import re
+import subprocess
+
+
+def is_package_exist_with_hashes(package_name,hash_of_file):
+    """ Check if any line in the file contains package with same hashes """
+    # Open the file in read only mode
+    line_with_hash = "{} \\\n    --hash=sha256:{}".format(package_name,hash_of_file)
+    with open('requirements.txt') as file:
+        # Read all lines in the file one by one
+        for line in file:
+            # For each line, check if line contains the package with same hashes
+            if line_with_hash in line:
+                return True
+    return False
+
+def is_package_exist_without_hashes(package_name):
+    """ Check if any line in the file contains package without same hash """
+    # Open the file in read only mode
+    line_with_hash = "{} \\\n".format(package_name) 
+    with open('requirements.txt') as file:
+        # Read all lines in the file one by one
+        for line in file:
+            # For each line, check if line contains the string
+            if line_with_hash in line:
+                return True
+    return False
+
+
+def add_package(package_name,hash_of_file):
+    with open('requirements.txt', 'a') as file:
+    # Append package with hashes at the end of file
+        line = "{} \\\n    --hash=sha256:{}\n".format(package_name,hash_of_file)
+        file.write(line)
+
+def append_requirement_file(package_name,hash_of_file):
+    if is_package_exist_with_hashes(package_name,hash_of_file):
+        return
+    else:
+        inputfile = open('requirements.txt').readlines()
+        write_file = open('requirements.txt','w')
+        for line in inputfile:
+            write_file.write(line)
+            current_line = "{} \\\n".format(package_name)
+            if current_line in line:
+                new_line = "    --hash=sha256:{} \\".format(hash_of_file)        
+                write_file.write(new_line + "\n") 
+        write_file.close()
+        
+
+def read_chunks(file, size=io.DEFAULT_BUFFER_SIZE):
+    """Yield pieces of data from a file-like object until EOF."""
+    while True:
+        chunk = file.read(size)
+        if not chunk:
+            break
+        yield chunk
+
+def hash_of_file(path):
+    # type: (str) -> str
+    """Return the hash digest of a file."""
+    with open(path, 'rb') as archive:
+        hash = hashlib.new('sha256')
+        for chunk in read_chunks(archive):
+            hash.update(chunk)
+    return hash.hexdigest()
+
+def main():
+    os.chdir(os.pardir) #go to etc
+    os.chdir(os.pardir)  # go to scancode-toolkit
+    cwd = os.getcwd()
+    print("Current working directory:", cwd) 
+    fp= open('requirements.txt','w') #Create empty file because it must exits before opening the file in read mode.
+    for subdir, dirs, files in os.walk(cwd+"/thirdparty"):
```

Comment:
You should accept an argument for the thirdparty directory. Also it would be likely easier to understand if you load all file paths at once and iterate after. You should also reuse the commonode.fileutils functions

Reply:
You talking about which function in fileutils?

Reply:
`resource_iter()`